staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -340,38 +343,29 @@ func (s *SpdyRoundTripper) proxyAuth(proxyURL *url.URL) string {
 // clients may call SpdyRoundTripper.Connection() to retrieve the upgraded
 // connection.
 func (s *SpdyRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
-	header := utilnet.CloneHeader(req.Header)
-	header.Add(httpstream.HeaderConnection, httpstream.HeaderUpgrade)
-	header.Add(httpstream.HeaderUpgrade, HeaderSpdy31)
-
-	var (
-		conn        net.Conn
-		rawResponse []byte
-		err         error
-	)
+	req = utilnet.CloneRequest(req)
+	req.Header.Add(httpstream.HeaderConnection, httpstream.HeaderUpgrade)
+	req.Header.Add(httpstream.HeaderUpgrade, HeaderSpdy31)
 
-	clone := utilnet.CloneRequest(req)
-	clone.Header = header
-	conn, err = s.Dial(clone)
+	success := false
+	conn, err := s.Dial(req)
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if !success {
+			conn.Close()
+		}
+	}()
 
-	responseReader := bufio.NewReader(
-		io.MultiReader(
```

Comment:
This is some cuft, no-op code. I think the intention was probably to handle the situation when buffered reader reads past the end of the response and those bytes need to be "unread" back into the connection. But the code doesn't handle that. This PR doesn't fix the problem, but it also doesn't change this behavior in any way. This bug was here since this code was initially added in 2015 and nobody noticed the bug.

Reply:
indeed is noop, but it seems was added to handled redirects https://github.com/kubernetes/kubernetes/commit/715d5d9c91c669cf33c0bf9a9c9d352c6c4228a6 , 

```
  174     if s.followRedirects {
  175         conn, rawResponse, err = utilnet.ConnectWithRedirects(req.Method, req.URL, header, req.Body, s, false)
```

and redirects is no longer used neither rawResponse :)